### PR TITLE
Fix equality check when other contains more keys

### PR DESCRIPTION
**Pull request by gagoman**
Test shows the issue.

Only Python version is affected.

P.S. I can not come up with more test cases, so I'm not sure that fix is final

### CODE REVIEW

File: multidict/_multidict_py.py [hunk 1]
```diff
@@ -120,6 +120,8 @@ def __eq__(self, other):
                 if i1 != i2 or v1 != v2:
                     return False
             return True
+        if len(self) != len(other):
```
Comment (webknjaz) [inline multidict/_multidict_py.py]: It must be possible to move this to before line `114` and remove check at line `117`. They are equal, so no need to have it copy-pasted in two places.

Comment (asvetlov) [reply multidict/_multidict_py.py]: I disagree.
Check from line 117 is faster than line 123: no python magic method `__len__` is involved.
Code duplication is very low, let's merge PR as is.

Comment (webknjaz) [reply multidict/_multidict_py.py]: If it's about speed, then I'd convert `len(self)` into `self._impl._items`

**Comment by asvetlov** [comment]
done

**Pull request merged**